packages/next-app/src/app/internalmanager/page.tsx

```diff
@@ -188,3 +153,100 @@ export default function Page() {
     </div>
   );
 }
+
+function TableRow({
+  token,
+  userAddress,
+  setTransaction,
+  chain,
+}: {
+  token: {
+    __typename?: "UserInternalBalance" | undefined;
+    token: `0x${string}`;
+    balance: string;
+  };
+  userAddress: `0x${string}`;
+  setTransaction: React.Dispatch<React.SetStateAction<ITransaction>>;
+  chain?: Chain;
+}) {
+  const [transferKind, setTransferKind] = useState<UserBalanceOpKind>();
+  const userBalanceOp = {
+    kind: transferKind as number,
+    asset: token.token,
+    amount: parseFixed(token.balance, tokenDictionary[token.token].decimals),
+    sender: userAddress as `0x${string}`,
+    recipient: userAddress as `0x${string}`,
+  };
+  const { config } = usePrepareVaultManageUserBalance({
+    args: [[userBalanceOp]],
+  });
+  const { data, write } = useVaultManageUserBalance(config);
+
+  function handleWithdraw() {
+    setTransferKind(UserBalanceOpKind.WITHDRAW_INTERNAL);
+    setTransaction((prev) => ({
+      ...prev,
+      status: TransactionStatus.WAITING_APPROVAL,
```

Comment:
wondering if we shouldn't use [useWaitForTransaction](https://wagmi.sh/react/hooks/useWaitForTransaction) to help us tracking the status of a submitted transaction